### ISSUE TITLE
API server connects to etcd over TLS

### ISSUE THREAD

**Issue by sym3tri** [coreos/tectonic-installer#233]
Blocked on upstream bootkube changes. @aaronlevy please comment when it's ready

**Comment by alexsomesan** [comment]
Assigning to @aaronlevy to give us green light when to move forward with this. Please assign back to me (or someone in tectonic) when ready.

**Comment by aaronlevy** [comment]
Because Tectonic runs its own render step - this shouldn't rely on any changes to bootkube (but this has been implemented on the bootkube end, so maybe can use that as reference steps (https://github.com/kubernetes-incubator/bootkube/pull/433).

The changes in rendering would mean:

1. bootstrap static api-server manifest (added in bootkube v0.4.0) needs to be rendered with etcd flags & have the etcd assets available on disk (temporarily): https://github.com/kubernetes-incubator/bootkube/blob/master/pkg/asset/internal/templates.go#L217-L272.
1. The self-hosted api-server manifest (daemonset) needs to have etcd flags that reference a file path that will be populated by a secret: https://github.com/kubernetes-incubator/bootkube/blob/master/pkg/asset/internal/templates.go#L174-L210
1. the api-server secret object needs to have all etcd assets included: https://github.com/kubernetes-incubator/bootkube/blob/master/pkg/asset/k8s.go#L90-L111

**Comment by s-urbaniak** [comment]
@sym3tri I believe this is a duplicate of #147, but I would like to verify that this is meant only for the "bring your own etcd" use case.

**Comment by sym3tri** [comment]
Yes, this is a dupe. Sorry.

**Comment by s-urbaniak** [comment]
closing in favor of #147
